nrich-validation/src/main/java/net/croz/nrich/validation/constraint/util/DateUtil.java

```diff
@@ -0,0 +1,65 @@
+package net.croz.nrich.validation.constraint.util;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.MonthDay;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.ZoneId;
+import java.time.chrono.ChronoLocalDate;
+import java.time.chrono.ChronoLocalDateTime;
+import java.time.chrono.ChronoZonedDateTime;
+import java.util.Calendar;
+import java.util.Date;
+
+public final class DateUtil {
```

Comment:
Not sure where did my comment end up so I'll put it here again take a look at DefaultCellValueConverter and try to do something similar it will avoid these if/else. You can also add a test for DateUtil